src/vol2bird.c

```diff
@@ -341,111 +386,124 @@ int main(int argc, char** argv) {
+        sd_vvp_thresh = &alldata.options.stdDevMinBird;
+        vcp = &alldata.misc.vcp;
+        wavelength = &alldata.options.radarWavelength;
+
+        int iRowProfile;
+        int iCopied = 0;
+
+        for (iRowProfile = 0; iRowProfile < nRowsProfile; iRowProfile++)
+        {
+            iCopied = iRowProfile * nColsProfile;
+            printf("%8s %.4s ", date, time);
+            printf("%4.f %6.2f %6.2f %7.2f %5.2f %5.1f %6.2f %s %6.2f %6.1f %6.2f %6.2f %5.f %5.f %5.f %5.f\n",
+                   profileBio[0 + iCopied],                                           // hght
+                   nanify(profileBio[2 + iCopied]), nanify(profileBio[3 + iCopied]),  // u,v
+                   nanify(profileBio[4 + iCopied]), nanify(profileBio[5 + iCopied]),  // w,ff
+                   nanify(profileBio[6 + iCopied]), nanify(profileAll[7 + iCopied]),  // dd,sd_vvp
+                   profileBio[8 + iCopied] == TRUE ? "TRUE" : "FALSE",                       // gap
+                   nanify(profileBio[9 + iCopied]), nanify(profileBio[11 + iCopied]), //  dbz,eta
+                   nanify(profileBio[12 + iCopied]), nanify(profileAll[9 + iCopied]), //  dens, dbz_all (dbzh)
+                   nanify(profileBio[10 + iCopied]), nanify(profileBio[13 + iCopied]),
+                   nanify(profileAll[10 + iCopied]), nanify(profileAll[13 + iCopied]));
+        }
+        profileAll = NULL;
+        profileBio = NULL;
+        free((void *)profileAll);
+        free((void *)profileBio);
 
         //}
     } // getter scope end
 
+    if (formatCSV){
+    
+        writeCSV(fileVpOut, &alldata, source, fileIn[0], date, time, volume);
 
-
+    } else {
     // ------------------------------------------------------------------- //
     //                 end of the getter example section                   //
-    // ------------------------------------------------------------------- //            
-        
-    //map vol2bird profile data to Rave profile object
-    mapDataToRave(volume, &alldata);
-    
-    //save rave profile to ODIM hdf5 file
-    if (fileVpOut != NULL){
-        int result;
-        result = saveToODIM((RaveCoreObject*) alldata.vp, fileVpOut); 
-        if (result == FALSE){
-            fprintf(stderr, "critical error, cannot write file %s\n", fileVpOut);
-            return -1;
+    // ------------------------------------------------------------------- //
+
+        // map vol2bird profile data to Rave profile object
+        mapDataToRave(volume, &alldata);
+
+        // RaveList_t* attnames = VerticalProfile_getAttributeNames(alldata.vp);
+        // RaveObjectList_t* attvalues = VerticalProfile_getAttributeValues(alldata.vp);
+        // printf("Number of names: %d\n", attnames);
+        // printf("Number of values: %d\n", attvalues);
+
+        // save rave profile to ODIM hdf5 file
+        if (fileVpOut != NULL)
+        {
+            int result;
+            result = saveToODIM((RaveCoreObject *)alldata.vp, fileVpOut);
```

Comment:
suggest inserting writeCSV function here, determining which format to write to based on extension of fileVpOut